realm/realm-library/src/androidTest/kotlin/io/realm/EmbeddedObjectsTest.kt

```diff
@@ -527,21 +528,155 @@ class EmbeddedObjectsTest {
     }
 
     @Test
-    @Ignore("Add in another PR")
     fun insertOrUpdate_deletesOldEmbeddedObject() {
-        TODO()
+        realm.executeTransaction { realm ->
+            val parent = EmbeddedSimpleParent("parent")
+            val originalChild = EmbeddedSimpleChild("originalChild")
+            parent.child = originalChild
+            realm.insert(parent)
+
+            assertEquals(1, realm.where<EmbeddedSimpleChild>().count())
+            val managedChild = realm.where<EmbeddedSimpleChild>()
+                    .equalTo("childId", "originalChild")
+                    .findFirst()
+            assertTrue(managedChild!!.isValid)
+
+            val newChild = EmbeddedSimpleChild("newChild")
+            parent.child = newChild
+            realm.insertOrUpdate(parent)
+            assertTrue(!managedChild.isValid)
+
+            assertEquals(1, realm.where<EmbeddedSimpleChild>().count())
+            val managedNewChild = realm.where<EmbeddedSimpleChild>()
+                    .equalTo("childId", "newChild")
+                    .findFirst()
+            assertEquals(managedNewChild!!.childId, "newChild")
+            realm.where<EmbeddedSimpleChild>()
+                    .equalTo("childId", "originalChild")
+                    .findAll()
+                    .size
+                    .also { oldChildCount ->
+                        assertEquals(0, oldChildCount)
+                    }
+        }
     }
 
     @Test
-    @Ignore("Add in another PR")
     fun insert_listWithEmbeddedObjects() {
-        TODO()
+        realm.executeTransaction { realm ->
+            val list = listOf<EmbeddedSimpleParent>()
+                    .plus(EmbeddedSimpleParent("parent1").apply { child = EmbeddedSimpleChild("child1") })
+                    .plus(EmbeddedSimpleParent("parent2").apply { child = EmbeddedSimpleChild("child2") })
+                    .plus(EmbeddedSimpleParent("parent3").apply { child = EmbeddedSimpleChild("child3") })
+                    .also { realm.insert(it) }
+
+            realm.where<EmbeddedSimpleParent>()
```

Comment:
Again, a bit funny organization. Would be nice with a cleaner separation of the query under test and the assertions, but can't immediately find a good suggestion.